### PR TITLE
snowflake_legacy 0.8.0 :snowflake:

### PR DESCRIPTION
**Pull request by lorepirri**
snowflake_legacy 0.8.0 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-4654]
- Upstream repository: private
- pypi: https://pypi.org/project/snowflake-legacy/0.8.0
- pypi inspector: https://inspector.pypi.io/project/snowflake-legacy/0.8.0

### Explanation of changes:

- new feedstock.
- patch `pyproject.toml` to fix paths.
- no `LICENSE` file.

[PKG-4654]: https://anaconda.atlassian.net/browse/PKG-4654?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/patches/0002-fix-pyproject.patch [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+diff --git a/pyproject.toml b/pyproject.toml
+index f1ab033..577d510 100644
+--- a/pyproject.toml
++++ b/pyproject.toml
+@@ -37,7 +37,7 @@ classifiers = [
+ dynamic = ['version']
+ 
+ [tool.hatch.version]
+-path = 'src/snowflake/_legacy/__init__.py'
++path = 'snowflake/_legacy/__init__.py'
```
Comment (JeanChristopheMorinPerso) [inline recipe/patches/0002-fix-pyproject.patch]: We should report all of these changes to Snowflake. This package seems pretty broken. This also includes the missing license in the sdist.

Comment (lorepirri) [reply recipe/patches/0002-fix-pyproject.patch]: Noted, thank you

File: recipe/patches/0001-add-snowflake-as-module.patch [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+---
+ snowflake/__init__.py | 1 +
+ 1 file changed, 1 insertion(+)
+ create mode 100644 snowflake/__init__.py
+
+diff --git a/snowflake/__init__.py b/snowflake/__init__.py
+new file mode 100644
+index 0000000..6e284c5
+--- /dev/null
++++ snowflake/__init__.py
```
Comment (JeanChristopheMorinPerso) [inline recipe/patches/0001-add-snowflake-as-module.patch]: I'm wondering if they omitted it intentionally or accidentally. Could we confirm this with them? Adding this file here could potentially break the namespace package.

Comment (lorepirri) [reply recipe/patches/0001-add-snowflake-as-module.patch]: Ok, sure

**Comment by JeanChristopheMorinPerso** [comment]
As discussed, we will need to make sure [snowflake.pth](https://inspector.pypi.io/project/snowflake-legacy/0.8.0/packages/16/cc/d3b5eb6d14c02b99bfeca54a2f8df6f5ad617359bcfe23c9a7d04ef25560/snowflake_legacy-0.8.0.tar.gz/snowflake_legacy-0.8.0/snowflake.pth) makes it into the package. Without this file, the package will not do what it's supposed to do.

File: recipe/patches/0001-add-snowflake-as-module.patch (hunk 2 above)
Comment (lorepirri) [reply recipe/patches/0001-add-snowflake-as-module.patch]: It needs to be omitted.

File: recipe/patches/0002-fix-pyproject.patch (hunk 1 above)
Comment (JeanChristopheMorinPerso) [reply recipe/patches/0002-fix-pyproject.patch]: As per SF, we should not add it. So we need to remove this part of the patch.

**Comment by lorepirri** [comment]
> As discussed, we will need to make sure [snowflake.pth](https://inspector.pypi.io/project/snowflake-legacy/0.8.0/packages/16/cc/d3b5eb6d14c02b99bfeca54a2f8df6f5ad617359bcfe23c9a7d04ef25560/snowflake_legacy-0.8.0.tar.gz/snowflake_legacy-0.8.0/snowflake.pth) makes it into the package. Without this file, the package will not do what it's supposed to do.

done, at the end, it was just about fixing the paths

File: recipe/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    - python
+    - hatchling
+    - pip
+  run:
+    - python
+
+test:
+  imports:
+    - snowflake
+    - snowflake._legacy
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:33]:
```suggestion
    - snowflake._legacy
    - snowflake.pth
```
Should we import `snowflake.pth` too?

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:33]: No, we can't import a pth file. It's "imported" at runtime, but you can't import it.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  requires:
+    - pip
+    - pytest
+  source_files:
+    - tests
+  commands:
+    - pip check
+    - pytest tests
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:41]: I would add a test for the existence of the pth file.

**Review by JeanChristopheMorinPerso — APPROVED** [review]

File: recipe/meta.yaml (hunk 3 above)
Comment (lorepirri) [reply recipe/meta.yaml:33]: @skupr-anaconda I've added a test for the existence of the pth file as suggested by @JeanChristopheMorinPerso

File: recipe/meta.yaml (hunk 4 above)
Comment (lorepirri) [reply recipe/meta.yaml:41]: done, thank you for the suggestion!

File: recipe/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    - python
+    - hatchling
+    - pip
+  run:
+    - python
+
+test:
+  imports:
+    - snowflake
+    - snowflake._legacy
```
Comment (Jrice1317) [inline recipe/meta.yaml:33]:
```suggestion
    - snowflake_legacy
```
Should the period be removed here as well?

**Review by Jrice1317 — APPROVED** [review]

File: recipe/meta.yaml (hunk 5 above)
Comment (Jrice1317) [reply recipe/meta.yaml:33]: Nvm, I see the package actually includes the `.` and we are only renaming it for our repo, correct?

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:33]: Nvm, I see the package actually includes the . and we are only renaming it for our repo, correct?
Yep. The installed python package is `snowflake._legacy`.

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Pull request merged**